xbmc/windowing/wayland/ShellSurfaceWebOSShell.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ShellSurfaceWebOSShell.h"
+
+#include "Registry.h"
+
+#include "utils/log.h"
+
+using namespace KODI::WINDOWING::WAYLAND;
+using namespace std::placeholders;
+
+using namespace wayland;
+
+/*
+  WebOS always is 1920x1080
+*/
+#define WEBOS_UI_WIDTH 1920
+#define WEBOS_UI_HEIGHT 1080
+
+CShellSurfaceWebOSShell::CShellSurfaceWebOSShell(IShellSurfaceHandler& handler,
+                                           CConnection& connection,
+                                           const wayland::surface_t& surface,
+                                           const std::string& title,
+                                           const std::string& class_)
+  : m_handler{handler}
+{
+  {
+    CRegistry registry{connection};
+    registry.RequestSingleton(m_shell, 1, 1);
+    registry.RequestSingleton(m_webos_shell, 1, 2);
+    registry.Bind();
+  }
+
+  m_shellSurface = m_shell.get_shell_surface(surface);
+
+  m_webos_shellSurface = m_webos_shell.get_shell_surface(surface);
+
+  m_webos_shellSurface.on_state_changed() = [this](std::uint32_t state) {
+    switch(state)
+    {
+      case (std::uint32_t)webos_shell_surface_state::fullscreen:
+        CLog::Log(LOGDEBUG, "webOS notification - Changed to full screen");
+        m_handler.OnConfigure(0, {WEBOS_UI_WIDTH, WEBOS_UI_HEIGHT}, m_surfaceState);
+        break;
+      case (std::uint32_t)webos_shell_surface_state::minimized:
+        CLog::Log(LOGDEBUG, "webOS notification - Changed to minimized");
+        break;
+    }
+  };
+
+  m_webos_shellSurface.on_close() = [this]() {
+    CLog::Log(LOGDEBUG, "webOS notification - Close notification received");
+    m_handler.OnClose();
+  };
+
+  const char *appId = NULL, *displayId = NULL;
```

Comment:
dont declare variables inline, also dont use NULL
see our code guidelines for initializations